src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VsCoreProjectSystemReferenceReader.cs

```diff
@@ -81,17 +82,30 @@ internal class VsCoreProjectSystemReferenceReader
 
                         var childProjectPath = reference3.SourceProject.GetFullProjectPath();
 
+
                         // Skip projects which have ReferenceOutputAssembly=false
-                        if (!string.IsNullOrEmpty(childProjectPath)
-                            && !excludedProjects.Contains(childProjectPath, StringComparer.OrdinalIgnoreCase))
+                        var reference6 = childReference as Reference6;
+                        var addProject = true;
+
+                        if (reference6 != null)
```

Comment:
You can delete line 87 with this:

```suggestion
                        if (childReference is Reference6 reference6)
```